# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/loader/WorkerLoadContext.h
# Commit: 0ae1fd421d4f
# Full Hash: 0ae1fd421d4fbb45b956df108f1cd13626eb3a3e
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-12-21 21:21:23
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement initial Static Module Loading for Workers; r=jonco
#   
#   This is the first pass of getting static module loading to work. This roughly implements
#   https://html.spec.whatwg.org/multipage/webappapis.html#fetch-a-worklet/module-worker-script-graph --
#   without some of the settings objects correctly set.
# ==============================================================================

diff -r 7770ec4717fd -r 0ae1fd421d4f dom/workers/loader/WorkerLoadContext.h
--- a/dom/workers/loader/WorkerLoadContext.h	Tue Dec 20 20:56:12 2022 +0000
+++ b/dom/workers/loader/WorkerLoadContext.h	Tue Dec 20 20:56:13 2022 +0000
@@ -85,6 +85,10 @@
     // We are importing a script from the worker via ImportScript. This may only
     // be a Classic script.
     ImportScript,
+    // We are importing a script from the worker via a Static Import. This may
+    // only
+    // be a Module script.
+    StaticImport,
     // We have an attached debugger, and these should be treated specially and
     // not like a main script (regardless of their type). This is not part of
     // the specification.